flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -131,6 +141,12 @@ public boolean scaleResource(
                 getVertexParallelismOverrides(
                         evaluatedMetrics.getVertexMetrics(), scalingSummaries));
 
+        if (tmHeapMemoryOpt.isPresent()) {
+            Configuration configOverrides = autoScalerStateStore.getConfigOverrides(context);
+            configOverrides.set(TaskManagerOptions.TASK_HEAP_MEMORY, tmHeapMemoryOpt.get());
```

Comment:
I wonder what will happen if this is too big and the user also defined other memory options like managed fraction etc. We may create a config that simply won't run / make sense